src/components/ShareModal/index.tsx

```diff
@@ -0,0 +1,120 @@
+'use client';
+
+import { useState, useRef, useEffect } from 'react';
+
+import { assignInlineVars } from '@vanilla-extract/dynamic';
+
+import { shareUserTestData } from '../../../public/shareUserTest';
+
+import useDebounce from 'hooks/useDebounce';
+
+import * as S from './index.css';
+import * as I from 'assets';
+import { ShareUserCard } from 'components';
+
+const ShareModal = () => {
+  const [inputValue, setInputValue] = useState<string>('');
+  const [isSelected, setIsSelected] = useState<string[]>([]);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (isSelected.length > 0) {
+      inputRef.current?.focus();
+    }
+  }, [isSelected]);
+
+  const debounceSerachUser = useDebounce(inputValue, 300);
+
+  const isUserSelected = (userId: string) => {
+    setIsSelected(prev => {
+      const prevSelectedUser = prev.includes(userId);
+
+      return prevSelectedUser
+        ? prev.filter(user => user !== userId)
+        : [...prev, userId];
+    });
```

Comment:
람다식과 삼항 연산자의 가장 큰 장점은 간결함이라고 생각합니다. 만약 중괄호를 사용한다면 if문으로 가독성을 올리는 편이 좋을 것 같습니다. 그게 아니라면 아래의 코드는 어떨까요?

```suggestion
    setIsSelected(prev => prev.includes(userId) ? prev.filter(user => user !== userId) : [...prev, userId]);
```

Reply:
저는 아래 승제님의 코드가 간결하고 보기도 좋아서 더 좋은것같습니다

Reply:
수정했습니다. 357fc90